CoordinateSystemsCore/src/points.jl

```diff
@@ -0,0 +1,148 @@
+abstract type AbstractPoint{CT<:CoordinateSystem, N, T} end
+@inline Base.eltype(::AbstractPoint{CT,N,T}) where {CT,N,T} = T
+@inline Base.ndims(::AbstractPoint{CT,N}) where {CT,N} = N
+@inline Base.:(==)(x::AbstractPoint, y::AbstractPoint) = with_coordinate_system(==, coordinate_system(x), x, y)
+@inline function Base.isapprox(x::AbstractPoint, y::AbstractPoint; kwargs...)
+    with_coordinate_system(coordinate_system(x), x, y) do p, q
```

Comment:
It strikes me that `with_coordinate_system` is very like `Base.promote` in function, if not in API. Perhaps that analogy is worth exploring?

Reply:
Yes! I had the same feeling, and that's why I'm thinking of making `AbstractPoint <: AbstractVector`, and then maybe it's possible to hide all the coordinate chart transition details under the scene.

Reply:
It could make sense to adopt a `promote`-like API (ie, return `p` and `q` rather than passing them to a callback) to make the use of this more imperative in style?

About `AbstractPoint <: AbstractVector`... maybe? I'm not sure what that implies!